'lit :: progress-bar.py' test is flaky

I thought we had a bug for this but couldn't find one. The test fails sometimes, especially on systems with high load. IIUC it prints one dot too many but I have no clue why.

```
******************** TEST 'lit :: progress-bar.py' FAILED ********************
Exit Code: 1

Command Output (stdout):
--
# RUN: at line 3
not env -u FILECHECK_OPTS "/tmp/portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit-python3_12/install/usr/bin/python3" /tmp/
portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit/lit.py -j1 --order=lexical -s Inputs/progress-bar > /tmp/portage/dev-pyth
on/lit-18.0.0_pre20231111/work/llvm/utils/lit/tests/Output/progress-bar.py.tmp.out
# executed command: not env -u FILECHECK_OPTS /tmp/portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit-python3_12/install/usr
/bin/python3 /tmp/portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit/lit.py -j1 --order=lexical -s Inputs/progress-bar
# RUN: at line 4
FileCheck < /tmp/portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit/tests/Output/progress-bar.py.tmp.out /tmp/portage/dev-py
thon/lit-18.0.0_pre20231111/work/llvm/utils/lit/tests/progress-bar.py
# executed command: FileCheck /tmp/portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit/tests/progress-bar.py
# .---command stderr------------
# | /tmp/portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit/tests/progress-bar.py:9:10: error: CHECK: expected string not fo
und in input
# | # CHECK: FAIL: progress-bar :: test-2.txt (2 of 4)
# |          ^
# | <stdin>:6:21: note: scanning from here
# | Testing: 0.. 10.. 20.
# |                     ^
# | <stdin>:7:1: note: possible intended match here
# | FAIL: progress-bar :: test-3.txt (3 of 4)
# | ^
# | 
# | Input file: <stdin>
# | Check file: /tmp/portage/dev-python/lit-18.0.0_pre20231111/work/llvm/utils/lit/tests/progress-bar.py
# | 
# | -dump-input=help explains the following input dump.
# | 
# | Input was:
# | <<<<<<
# |            1: -- Testing: 4 tests, 1 workers -- 
# |            2: Testing:  
# |            3: FAIL: progress-bar :: test-1.txt (1 of 4) 
# |            4: Testing: 0.. 10 
# |            5: FAIL: progress-bar :: test-2.txt (2 of 4) 
# |            6: Testing: 0.. 10.. 20. 
# | check:9'0                         X~ error: no match found
# |            7: FAIL: progress-bar :: test-3.txt (3 of 4) 
# | check:9'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# | check:9'1     ?                                          possible intended match
# |            8: Testing: 0.. 10.. 20.. 30.. 40.. 50.. 60 
# | check:9'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |            9: FAIL: progress-bar :: test-4.txt (4 of 4) 
# | check:9'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |           10: Testing: 0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.. 80.. 90..  
# | check:9'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |           11: ******************** 
# | check:9'0     ~~~~~~~~~~~~~~~~~~~~~
# |           12: Failed Tests (4): 
# | check:9'0     ~~~~~~~~~~~~~~~~~~
# |            .
# |            .
# |            .
# | >>>>>>
# `-----------------------------
# error: command failed with exit status: 1

--

********************
```

I've been hitting this recently, so I've put up a PR to fix this.

I got:
```
Command Output (stdout):
  --
  # RUN: at line 3
  not env -u FILECHECK_OPTS "/usr/bin/python3" /home/jontha01/llvm-project/llvm/utils/lit/lit.py -j1 -D use_normalized_slashes=0
  --order=lexical -s Inputs/progress-bar > /data_nvme1n1/jontha01/llvm-project/rel/utils/lit/tests/Output/progress-bar.py.tmp.out
  # executed command: not env -u FILECHECK_OPTS /usr/bin/python3 /home/jontha01/llvm-project/llvm/utils/lit/lit.py -j1 -D
  use_normalized_slashes=0 --order=lexical -s Inputs/progress-bar
  # RUN: at line 4
  FileCheck < /data_nvme1n1/jontha01/llvm-project/rel/utils/lit/tests/Output/progress-bar.py.tmp.out /data_nvme1n1/jontha01/llvm-project/rel/
  utils/lit/tests/progress-bar.py
  # executed command: FileCheck /data_nvme1n1/jontha01/llvm-project/rel/utils/lit/tests/progress-bar.py
  # .---command stderr------------
  # | /data_nvme1n1/jontha01/llvm-project/rel/utils/lit/tests/progress-bar.py:13:10: error: CHECK: expected string not found in input
  # | # CHECK: FAIL: progress-bar :: test-4.txt (4 of 4)
  # |          ^
  # | <stdin>:10:46: note: scanning from here
  # | Testing: 0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.. 80.. 90..
  # |                                              ^
  # | <stdin>:16:2: note: possible intended match here
  # |  progress-bar :: test-4.txt
  # |  ^
  # |
  # | Input file: <stdin>
  # | Check file: /data_nvme1n1/jontha01/llvm-project/rel/utils/lit/tests/progress-bar.py
  # |
  # | -dump-input=help explains the following input dump.
  # |
  # | Input was:
  # | <<<<<<
  # |             .
  # |             .
  # |             .
  # |             5: FAIL: progress-bar :: test-2.txt (2 of 4)
  # |             6: Testing: 0.. 10.. 20.. 30.. 40.. 50..
  # |             7: FAIL: progress-bar :: test-3.txt (3 of 4)
  # |             8: Testing: 0.. 10.. 20.. 30.. 40.. 50.. 60..
  # |             9: FAIL: progress-bar :: test-4.txt (4 of 4)
  # |            10: Testing: 0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.. 80.. 90..
  # | check:13'0                                                  X~~~~~~~~~~~~~ error: no match found
  # |            11: ********************
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~~
  # |            12: Failed Tests (4):
  # | check:13'0     ~~~~~~~~~~~~~~~~~~
  # |            13:  progress-bar :: test-1.txt
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |            14:  progress-bar :: test-2.txt
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |            15:  progress-bar :: test-3.txt
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |            16:  progress-bar :: test-4.txt
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # | check:13'1      ?                           possible intended match
  # |            17:
  # | check:13'0     ~
  # |            18:
  # | check:13'0     ~
  # |            19: Testing Time: 0.03s
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~
  # |            20:
  # | check:13'0     ~
  # |            21: Total Discovered Tests: 4
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |            22:  Failed: 4 (100.00%)
  # | check:13'0     ~~~~~~~~~~~~~~~~~~~~~
  # | >>>>>>
  # `-----------------------------
  # error: command failed with exit status: 1
```

--- a/llvm/utils/lit/tests/progress-bar.py
+++ b/llvm/utils/lit/tests/progress-bar.py
@@ -3,12 +3,19 @@
 # RUN: not %{lit} -s %{inputs}/progress-bar > %t.out
 # RUN: FileCheck < %t.out %s
 #
-# CHECK: Testing:
-# CHECK: FAIL: progress-bar :: test-1.txt (1 of 4)
-# CHECK: Testing:  0.. 10.. 20
-# CHECK: FAIL: progress-bar :: test-2.txt (2 of 4)
-# CHECK: Testing:  0.. 10.. 20.. 30.. 40..
-# CHECK: FAIL: progress-bar :: test-3.txt (3 of 4)
-# CHECK: Testing:  0.. 10.. 20.. 30.. 40.. 50.. 60.. 70
-# CHECK: FAIL: progress-bar :: test-4.txt (4 of 4)
-# CHECK: Testing:  0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.. 80.. 90..
+# CHECK: -- Testing: 4 tests, 1 workers --
+# CHECK-NEXT: Testing:
+# CHECK-NEXT: FAIL: progress-bar :: test-1.txt (1 of 4)
+# CHECK-NEXT: Testing:
+# CHECK-NEXT: FAIL: progress-bar :: test-2.txt (2 of 4)
+# CHECK-NEXT: Testing:
+# CHECK-NEXT: FAIL: progress-bar :: test-3.txt (3 of 4)
+# CHECK-NEXT: Testing:
+# CHECK-NEXT: FAIL: progress-bar :: test-4.txt (4 of 4)
+# CHECK-NEXT: Testing:
+# CHECK-NEXT: ********************
+# CHECK-NEXT: Failed Tests (4):
+# CHECK-NEXT:   progress-bar :: test-1.txt
+# CHECK-NEXT:   progress-bar :: test-2.txt
+# CHECK-NEXT:   progress-bar :: test-3.txt
+# CHECK-NEXT:   progress-bar :: test-4.txt

--- a/llvm/utils/lit/tests/progress-bar.py
+++ b/llvm/utils/lit/tests/progress-bar.py
@@ -12,7 +12,7 @@
 # CHECK-NEXT: FAIL: progress-bar :: test-3.txt (3 of 4)
 # CHECK-NEXT: Testing:
 # CHECK-NEXT: FAIL: progress-bar :: test-4.txt (4 of 4)
-# CHECK-NEXT: Testing:
+# CHECK-NEXT: Testing:  0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.. 80.. 90..
 # CHECK-NEXT: ********************
 # CHECK-NEXT: Failed Tests (4):
 # CHECK-NEXT:   progress-bar :: test-1.txt

